### PR TITLE
Connect Authentication 🔑

### PR DESCRIPTION
**Pull request by TheLukaszNs**
## Summary
Added authentication connecting with the backend auth services. The token is stored in SecureStore - the current user is provided via the new `AuthContext` context.

❗ Note: you need to create a `.env` file in the root of the app module with the following content:
```env
API_URL="<HOST IP>"
API_PORT="<HOST API PORT>"
```
The app will throw an error if you don't complete this step.

### CODE REVIEW

File: app/.gitignore [hunk 1]
```diff
@@ -12,3 +12,5 @@ web-build/
 
 # macOS
 .DS_Store
+
+.env
```
Comment (Pichi00) [inline app/.gitignore]:
```suggestion
.env

```

Comment (Pichi00) [reply app/.gitignore]: EOF

**Review by Pichi00 — CHANGES_REQUESTED** [review]

File: app/.gitignore (hunk 1 above)
Comment (TheLukaszNs) [reply app/.gitignore]: Rly düd? 🥶

**Review by inux71 — APPROVED** [review]
Approved! 👍

**Pull request merged**